rlclientlib/utility/context_helper.cc

```diff
@@ -144,4 +145,35 @@ namespace reinforcement_learning { namespace utility {
     }
     return error_code::success;
   }
+
+  int get_slot_ids_or_generate_from_index(const char* context, const ContextInfo::index_vector_t& slots, std::vector<std::string>& slot_ids, i_trace* trace, api_status* status)
+  {
+    // if slot id missing, use the slot index as the slot id
+    for(size_t i = 0; i < slots.size(); i++)
+    {
+      try {
+        rj::Document obj;
+        obj.Parse(std::string(context + slots[i].first, slots[i].second).c_str());
+
+        if (obj.HasParseError()) {
+          RETURN_ERROR_LS(trace, status, json_parse_error) << "JSON parse error: " << rj::GetParseError_En(obj.GetParseError()) << " (" << obj.GetErrorOffset() << ")";
+        }
+
+        const rj::Value::ConstMemberIterator& itr = obj.FindMember(slot_id);
+        if (itr != obj.MemberEnd() && itr->value.IsString()) {
+          slot_ids.emplace_back(std::string(itr->value.GetString()));
+        }
+        else {
+          slot_ids.emplace_back(std::to_string(slot_ids.size()));

```

Comment:
check for duplicates with provided slot ids?

Reply:
My understanding is that we don't check for this at this point, and assume that user provided slot ids won't be identical here